### ISSUE TITLE
Is AUROC from lightning metrics package intentionally different from the sklearn's?

### ISSUE THREAD

**Issue by stllfe** [PyTorchLightning/pytorch-lightning#2621]
## ❓ Questions and Help

Faced this weird behaviour trying to figure out why my model's auroc metric sucks balls deep, while all the others (e. g. accuracy, sensitivity, specificity) are relatively high.

Given two exact same tensors of targets and model's outputs, sklearn metric version outputs high score as intended, whereas plain tensors version gives me something like the opposite of that score. For example, if sklearn metric outputs a roc score of 0.975, tensor metric gives me 0.025. 

#### What is your question?
Is it intentional or some sort of a bug, or am I dumb ass? Sorry for the latter though... I have only used sklearn metrics before, so I kinda got used to them.

#### Code
```python
>>> import torch
>>> from pytorch_lightning.metrics.classification import AUROC
>>> from pytorch_lightning.metrics.sklearns import AUROC as SKAUROC

>>> auroc = AUROC()
>>> sk_auroc = SKAUROC()

>>> outs = torch.Tensor([[0.8, 0.2], [0.8, 0.2], [0.9, 0.1], [0.1, 0.9]])
>>> targets = torch.Tensor([0, 0, 0, 1])
>>> auroc(pred=outs, target=targets)
tensor(0.)

>>> sk_auroc(y_score=outs[:, 1], y_pred=targets)
tensor(1.)
```

#### What's your environment?

```
* CUDA:
        - GPU:
                - GeForce GTX 1080 Ti
        - available:         True
        - version:           10.2
* Packages:
        - numpy:             1.18.5
        - pyTorch_debug:     False
        - pyTorch_version:   1.5.0
        - pytorch-lightning: 0.8.5
        - tensorboard:       2.2.2
        - tqdm:              4.45.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:         x86_64
        - python:            3.7.5
        - version:           #107-Ubuntu SMP Thu Jun 4 11:27:52 UTC 2020
```

**Comment by rohitgr7** [comment]
Try: `auroc(pred=outs[:, 1], target=targets)`, since it takes class `0` by default from predictions.
https://github.com/PyTorchLightning/pytorch-lightning/blob/7b4db3045dcc9e6bb0b66e409b25bb2c7fa378f0/pytorch_lightning/metrics/functional/classification.py#L490-L492

**Comment by stllfe** [comment]
@rohitgr7 thanks, bro, haven't even thought of that!

**Comment by stllfe** [comment]
As a last word of suggestion, it should probably raise some error just like the sklearn's does in case of multiple dimensions or provide some argument for user to decide. Reducing dimensions of predictions this way is a bit counterintuitive imo... 

In fact, I haven't thought of that due to unified metrics API. Now it seems weird to me to separate logic between, say, accuracy calculation and auroc, considering the former allows to change argmax dim transparently.
